### ISSUE TITLE
Augment flag keys in ".flags" type definition

### ISSUE THREAD

**Issue by voxpelli** [sindresorhus/meow#155]
Unfortunately this doesn't seems doable yet, but it is something that is discussed here it seems: https://github.com/Microsoft/TypeScript/issues/12754

So currently `.flags['foo-bar']` will have the same type as `.unnormalizedFlags['foo-bar']`, whereas only the latter should have that, the former should have it's key augmented into `.flags.fooBar` or at least have the errenous `foo-bar` key removed.

I believe none of that is possible right now, but wanted to open an issue and track that it would be preferable, as that can then eg. tie into the discussions within TypeScript itself.

**Comment by voxpelli** [comment]
Correcting myself: `.unnormalizedFlags` shouldn't have kebabcased keys either.

**Comment by voxpelli** [comment]
Closing this as this somewhat duplicates: https://github.com/sindresorhus/meow/issues/165#issuecomment-706729260

Just a note, it may soon be possible to do something about this as the [TypeScript 4.1 beta supports](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1-beta/#template-literal-types):

```typescript
type Getters<T> = {
    [K in keyof T as `get${capitalize K}`]: () => T[K]
};

interface Person {
    name: string;
    age: number;
    location: string;
}

type LazyPerson = Getters<Person>;
```

If there would be a `Split<"foo-bar", "-">` (they are switching to having `capitalize K` be `Capitalize<K>` in https://github.com/microsoft/TypeScript/pull/40580) then maybe one could rewrite it as:

```typescript
type CamelCase<K extends string> = `${Split<K, "-">[0]}${Capitalize<Split<K, "-">[1] | "">}

type CamelCasedProps<T> = {
    [K in keyof T as CamelCase<K>]: () => T[K]
};

interface KebabCased {
    "foo-bar": string;
    foo: number;
}

type CamelCased = CamelCasedProps<KebabCased>;
```

**Comment by voxpelli** [comment]
Actually, it is already possible: https://github.com/microsoft/TypeScript/issues/40710#issuecomment-718677449

**Comment by sindresorhus** [comment]
@voxpelli Should we reopen this?

**Comment by voxpelli** [comment]
@sindresorhus Yeah, lets do it.

So, this would rely on https://github.com/sindresorhus/type-fest/issues/134 I'm thinking.

+ it would require TypeScript 4.1, which I'm not really sure when/how one can reliably require? At least would have to wait for eg. VS Code to have shipped with it (and of course for TS 4.1 to itself have shipped)

Unless of course one can publish these new types in a way that works with older TS versions, but I doubt it

**Comment by sindresorhus** [comment]
> it would require TypeScript 4.1, which I'm not really sure when/how one can reliably require? At least would have to wait for eg. VS Code to have shipped with it (and of course for TS 4.1 to itself have shipped)

I'm ok with requiring TS 4.1 a couple of months after it has shipped.
